tools/clang/lib/AST/Expr.cpp

```diff
@@ -833,7 +833,7 @@ StringLiteral *StringLiteral::Create(const ASTContext &C, StringRef Str,
                                      StringKind Kind, bool Pascal, QualType Ty,
                                      const SourceLocation *Loc,
                                      unsigned NumStrs) {
-  assert(C.getAsConstantArrayType(Ty) &&
+  assert(hlsl::IsStringLiteralType(Ty) &&
```

Comment:
perhaps use DXASSERT for our asserts.

Reply:
This file is mostly original clang code so it is probably best to keep the original assert here (I'll revert the hlsl::InStringLiteralType() check and use the original one). In other cases I'll use DXASSERT where appropriate.

Reply:
Makes sense.